src/main/java/com/monumental/services/ModelService.java

```diff
@@ -66,16 +64,21 @@
 
         Session session = this.sessionFactoryService.getFactory().openSession();
         Transaction transaction = null;
-        List<T> records = new ArrayList<>();
+        List<T> records;
 
         try {
             transaction = session.beginTransaction();
+            String tableName = this.getModelClass().getName();
             if (ids == null) {
-                records = session.createQuery("FROM " + this.getModelClass().getName()).list();
+                records = session.createQuery("FROM " + tableName).list();
+            } else if (ids.size() == 1) {
+                records = new ArrayList<>();
+                records.add((T) session.get(this.getModelClass(), ids.get(0)));
             } else {
-                for (Integer id : ids) {
-                    records.add((T) session.get(this.getModelClass(), id));
-                }
+                records = session.createQuery(
+                    "FROM " + tableName +
+                    "WHERE id IN (:ids)"
+                ).setParameter("ids", ids).list();
```

Comment:
This theoretically makes querying for lists of ids more efficient since it's 1 query instead of N